flink-tests/src/test/java/org/apache/flink/test/scheduling/SpeculativeSchedulerITCase.java

```diff
@@ -184,6 +185,23 @@ public void testSpeculativeSlowSink() throws Exception {
         assertThat(DummyCommitter.foundSpeculativeWriter).isTrue();
     }
 
+    @Test
+    public void testNonSpeculativeSlowSinkFunction() throws Exception {
+        executeJob(this::setupNonSpeculativeSlowSinkFunction);
+        waitUntilJobArchived();
+
+        checkResults();
+    }
+
+    @Test
+    @Timeout(30)
```

Comment:
`Timeout` should be avoided for Flink tests. See https://flink.apache.org/contributing/code-style-and-quality-common.html#avoid-timeouts-in-junit-tests.